csrc/device_lower/utils.h

```diff
@@ -304,6 +304,10 @@ bool isExtentEqualToMaxParallelTypeExtent(const IterDomain* id);
 //! indexing special items in shared memory, like mbarrier.
 NVF_API Val* u32IndexScalarSmemTv(TensorView* tv);
 
+//! Get the uint32_t index of a TensorIndex. This is usually used for
+//! initializing a pipeline of mbarriers.
+NVF_API Val* u32IndexScalarSmemTv(kir::TensorIndex* index);
```

Comment:
Why is this `NVF_API`?

Reply:
It feels unnecessary but this signature is derived from `NVF_API Val* u32IndexScalarSmemTv(TensorView* tv);`

Reply:
I'll try this change in a separate PR.